code/processes/wdb.q

```diff
@@ -372,13 +381,20 @@ endofdaysort:{[dir;pt;tablist;writedownmode;mergelimits;hdbsettings]
 
 /-function to send reload message to rdbs/hdbs
 reloadproc:{[h;d;ptype]
-	.wdb.countreload:count[raze .servers.getservers[`proctype;;()!();1b;0b]each reloadorder];
-	$[eodwaittime>0;
-		{[x;y;ptype].[{neg[y]@x};(x;y);{[ptype;x].lg.e[`reloadproc;"failed to reload the ",string[ptype]];'x}[ptype]]}[({@[`. `reload;x;()]; (neg .z.w)(`.wdb.handler;1b); (neg .z.w)[]};d);h;ptype];
-		@[h;(`reload;d);{[ptype;e] .lg.e[`reloadproc;"failed to reload the ",string[ptype],".  The error was : ",e]}[ptype]]
-	];
-	.lg.o[`reload;"the ",string[ptype]," has been successfully reloaded"];
-	}
+        .wdb.countreload:count[raze .servers.getservers[`proctype;;()!();1b;0b]each reloadorder];
+        $[eodwaittime>0;
+                {[x;y;ptype].[{neg[y]@x};(x;y);{[ptype;x].lg.e[`reloadproc;"failed to reload the ",string[ptype]];'x}[ptype]]}
+                        [({@[`. `reload;x;
+	/-if process fails to reload message with error entered into .wdb.d and  output error to process error log
+                                {[ptype;e](neg .z.w)(`.wdb.handler;(ptype;0b;`$"failed with error: ",e));
+                                .lg.e[`reloadproc;"failed to reload ",string[ptype]," from .wdb.reloadproc call. The error was : ",e]}[y]];
```

Comment:
Is this function definitely correct? It looks to me like if the reload fails then it will 

a) print that the reload has failed and send a error result back to the wdb
b) because the failure is caught, it will then send a pass result as well

Reply:
So at the moment it does as you have stated above - I have edited the .wdb.handler function to try and mitigate the problems with this. The handler function currently looks like this:
```
handler:{
        /-insert process reload outcome into .wdb.reloadstatus
        if[not .z.w in  key .wdb.reloadstatus;
                .wdb.reloadstatus[.z.w]:x];
        if[(not .wdb.reloadstatus[.z.w]`status);
                .lg.o[`reload;"the ", string[.wdb.reloadstatus[.z.w]`process]," process reload has ", string[.wdb.reloadstatus[.z.w]`result]]];
        if[(.proc.cp[]>.wdb.timeouttime) or (count[.wdb.reloadstatus]=.wdb.countreload);
                .lg.o[`handler;"releasing processes"];
                .lg.o[`reload;string[count select from .wdb.reloadstatus where status=1]," out of ", string[count .wdb.reloadstatus]," processes successfully reloaded"];
                .wdb.flushend[]];
        /-delete contents from .wdb.reloadstatus when reloads completed
        if[.wdb.reloadcomplete;delete from `.wdb.reloadstatus];
        };
```
I added this if statement at the start of the function when inserting the success/fail message that is passed as an argument when the handler function is called from .wdb.reloadproc:
```
 if[not .z.w in  key .wdb.reloadstatus;
                .wdb.reloadstatus[.z.w]:x];
```
In the case where a reload fails the .wdb.handler function will be first called with the failed message and that if statement will execute adding the message to .wdb.reloadstatus. When the following success message is called it shouldn't execute this if statement as the process handle is already a key in the table. 

The issue then if the final final process to reload fails:
The error message will be added to .wdb.reloadstatus and the count of the .wdb.reloadstatus will = the count of the number of processes to execute and this if statement will execute:
```
   if[(.proc.cp[]>.wdb.timeouttime) or (count[.wdb.reloadstatus]=.wdb.countreload);
                .lg.o[`handler;"releasing processes"];
                .lg.o[`reload;string[count select from .wdb.reloadstatus where status=1]," out of ", string[count .wdb.reloadstatus]," processes successfully reloaded"];
                .wdb.flushend[]];
```
as will this final if statement which will clear out the table
```
  if[.wdb.reloadcomplete;delete from `.wdb.reloadstatus];
```
and if the final reload fails the following success message will be inserted into the .wdb.reloadstatus - however as the reload has been completed the final if statement will execute again clearing out the table. This approach is not ideal and I'm working on a simpler approach now.